### PR TITLE
ffmpeg: include svt-av1 in small variant

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

svt-av1 is the fastest av1 encoder available. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ju1m — APPROVED** [review]
Thanks!
Indeed SVT-AV1 is the fastest AV1 software encoder; in the news: [SVT-AV1 1.6 Squeezes Out Even More Performance For CPU-Based AV1 Encoding](https://www.phoronix.com/news/SVT-AV1-1.6) (19 June 2023).

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 1]
```diff
@@ -79,7 +78,7 @@
 , withSrt ? withHeadlessDeps # Secure Reliable Transport (SRT) protocol
 , withSsh ? withHeadlessDeps # SFTP protocol
 , withSvg ? withFullDeps # SVG protocol
-, withSvtav1 ? withFullDeps && !stdenv.isAarch64 # AV1 encoder/decoder (focused on speed and correctness)
+, withSvtav1 ? withSmallDeps && !stdenv.isAarch64 # AV1 encoder/decoder (focused on speed and correctness)
```
Comment (Atemu) [inline pkgs/development/libraries/ffmpeg/generic.nix:81]:
```suggestion
, withSvtav1 ? withHeadlessDeps && !stdenv.isAarch64 # AV1 encoder/decoder (focused on speed and correctness)
```

Headless systems also benefit from svt-av1 support. There's nothing "headed" about it.

File: pkgs/development/libraries/ffmpeg/generic.nix [hunk 2]
```diff
@@ -33,7 +33,6 @@
 , withBzlib ? withHeadlessDeps
 , withCaca ? withFullDeps # Textual display (ASCII art)
 , withCelt ? withFullDeps # CELT decoder
-, withCrystalhd ? withFullDeps
```
Comment (Atemu) [inline pkgs/development/libraries/ffmpeg/generic.nix:36]: Nit: Separate commit. Makes a potential revert easier.

**Review by Atemu — CHANGES_REQUESTED** [review]
Can't currently test this as I've only got an aarch64 machine with me but the change in general gets my approval.

I'd like to know what the closure size delta is though.

**Comment by SuperSandro2000** [comment]
> I'd like to know what the closure size delta is though.

before:
```
 ➜ nix path-info -Sh ./result-bin
/nix/store/n6qkjm9x0r6k3fxalzcpf3jpa60mr77c-ffmpeg-5.1.3-bin     574.6M
```

after
```
 ➜ nix path-info -Sh ./result-bin
/nix/store/qmihimgph30d9dcrpnk8sl6bkqpwmxzk-ffmpeg-5.1.3-bin     584.4M
```

**Comment by ju1m** [comment]
Seeing the number of rebuilds, maybe it should be redirected to the `staged` branch instead of `master`.

**Comment by Atemu** [comment]
Yes, please target staging and resolve the merge conflict.

**Comment by SuperSandro2000** [comment]
done

**Review by Atemu — APPROVED** [review]
Diff LGTM.

**Comment by Atemu** [comment]
There's a space char too many in your commit title.

**Comment by SuperSandro2000** [comment]
ups, fixed that

**Pull request merged**